builtin/common/serialize.lua

```diff
@@ -147,6 +147,8 @@ function core.serialize(x)
 				end
 			end
 			return "{"..table.concat(vals, ", ").."}"
+		elseif tp == "userdata" and x.to_string then
+			return string.format("%q", x:to_string())
```

Comment:
If you do a `deserialize(serialize(table))`, some userdata values are converted into strings, this is a little weird.

Reply:
Ah your right there's probably a better way of doing this I will close this pr for now.